clean-data/clean-data.R

```diff
@@ -0,0 +1,700 @@
+    jobRoleOtherNo <- part1 %>% filter(is.na(JobRoleInterestOther))
+    cleanPart1 <- jobRoleOtherNo %>% bind_rows(jobRoleOtherYes)
+
+    ## Change uncertain job roles to "Undecided"
+    ##  Adapted from: http://stackoverflow.com/a/26766945
+    undecidedWords <- c("not sure", "don't know", "not certain",
+                        "unsure", "dont know", "undecided",
+                        "no preference", "not", "any", "no idea")
+    idx <- cleanPart1 %>% select(JobRoleInterestOther) %>%
+        mutate_each(
+            funs(grepl(paste(undecidedWords, collapse = "|"),
+                       .,
+                       ignore.case = TRUE)
+                 )
+            ) %>% unlist(use.names = FALSE)
+    undecidedData <- cleanPart1 %>% filter(idx) %>%
+        mutate(JobRoleInterestOther = "Undecided")
+    cleanPart1 <- cleanPart1 %>% filter(!idx) %>% bind_rows(undecidedData)
+
+    ## Normalize cyber security interests to "Cyber Security"
+    ##  e.g. "Cyber security" == "Cybersercurity"
+    cyberIdx <- cleanPart1 %>% select(JobRoleInterestOther) %>%
+        mutate_each(funs(grepl("cyber", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    cyberData <- cleanPart1 %>% filter(cyberIdx) %>%
+        mutate(JobRoleInterestOther = "Cyber Security")
+    cleanPart1 <- cleanPart1 %>% filter(!cyberIdx) %>% bind_rows(cyberData)
+
+
+    # Clean expected earnings column
+    cleanPart1 <- clean_expected_earnings(cleanPart1)
+
+
+    # Code Events Other
+
+    ## Convert coding events to binary/boolean
+    codeResources <- cleanPart1 %>%
+        select(starts_with("CodeEvent"), -CodeEventOther) %>%
+        mutate_each(funs(ifelse(!is.na(.), "1", NA)))
+    cleanPart1 <- cleanPart1 %>%
+        select(-starts_with("CodeEvent"), CodeEventOther) %>%
+        bind_cols(codeResources)
+
+    ## Title case other coding events
+    codingEvents <- cleanPart1 %>% filter(!is.na(CodeEventOther)) %>%
+        mutate(CodeEventOther = simple_title_case(CodeEventOther))
+    codeEventsElse <- cleanPart1 %>% filter(is.na(CodeEventOther))
+    cleanPart1 <- codeEventsElse %>% bind_rows(codingEvents)
+
+    ## Normalize variations of "meetup" to "Meetups"
+    meetupWords <- c("meetup", "meet")
+    cleanPart1 <- normalize_text(inData = cleanPart1,
+                   columnName = "CodeEventOther",
+                   searchTerms = meetupWords,
+                   replaceWith = "Meetup(s)")
+
+    ## Normalize FreeCodeCamp
+    fccWords <- c("fcc", "freecodecamp", "free code camp")
+    cleanPart1 <- normalize_text(inData = cleanPart1,
+                                 columnName = "CodeEventOther",
+                                 searchTerms = fccWords,
+                                 replaceWith = "Free Code Camp")
+
+    ## Normalize variations of "None"
+    nones <- c("non", "none", "haven't", "havent", "not", "nothing",
+               "didn't", "n/a", "\bna\b", "never", "nil", "nope")
```

Comment:
That's actually very interesting, because first 1500 respondents have lots of "none" answers, but then it vanished almost totally. Did you forget to add "none" option from the start?